app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt

```diff
@@ -3,17 +3,19 @@ package com.example.criminalintentapp
 import android.os.Bundle
 import android.text.Editable
 import android.text.TextWatcher
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
 import android.widget.CheckBox
 import android.widget.EditText
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentResultListener
 import androidx.lifecycle.ViewModelProvider
 import java.util.UUID
 
-class CrimeFragment : Fragment() {
+class CrimeFragment : Fragment(), FragmentResultListener {
```

Comment:
Just as in CrimeListFragment, I would pass the layout to the `Fragment()` constructor